File: sdk/java/nmidsdk/worker/src/main/java/com/nmidsdk/worker/Function.java
package com.nmidsdk.worker;

/**
 * A registered worker function. Mirrors sdk/node/lib/worker/function.js.
 * <p>
 * The function receives the decoded job {@link Response} and must return the
 * ret payload (typically a msgpack-encoded RetStruct via
 * {@link com.nmidsdk.model.utils.Utils#buildRetStruct}). It may throw to
 * signal failure.
 * <p>
 * Implementations run on the worker's job thread pool, so blocking work
 * (e.g. calling an external HTTP API) is fine and will not block the reader
 * thread that receives jobs from the server.
 */
@FunctionalInterface
public interface Function {
    byte[] run(Response resp) throws Exception;
}

File: sdk/node/lib/worker/function.js
'use strict';

// A registered worker function. Mirrors sdk/python3/nmidsdk/worker/function.py.
//
// jobFunc signature: (resp: WorkerResponse) => Buffer | Promise<Buffer>
// The function receives the decoded job response and must return the ret
// payload (typically a msgpack-encoded RetStruct). It may throw to signal
// failure; returning a Promise is supported.

class Function {
  constructor(jobFunc, fname) {
    this.func = jobFunc;
    this.funcName = fname;
  }
}

module.exports = Function;
